gitlabform/processors/project/files_processor.py

```diff
@@ -17,27 +17,57 @@
 
 
 class FilesProcessor(AbstractProcessor):
+    def pre_protection_check(self, project_and_group: str, branch: str, desired_protected_state: bool, desired_push_access_level: str, desired_merge_access_level: str):
+        try:
+            branch_info = self.gitlab.get_branch(project_and_group, branch)
+            current_protected_state = branch_info["protected"]
+
+            # Check if branch needs unprotecting
+            needs_unprotect = (
+                current_protected_state 
+                and (
+                    branch_info["push_access_levels"][0]["access_level_description"] != desired_push_access_level 
+                    or branch_info["merge_access_levels"][0]["access_level_description"] != desired_merge_access_level
+                )
+            )
+
+            # If the branch is unprotected or should remain unprotected, skip unprotection.
+            if not current_protected_state and not desired_protected_state:
+                print(f"Branch '{branch}' is already unprotected and should remain so. Skipping changes.")
+                return False
+
+            if needs_unprotect:
+                print(f"Branch '{branch}' needs changes in protection settings.")
+                return True
+            else:
+                print(f"No changes needed on branch '{branch}'.")
+                return False
+
+        except NotFoundException:
+            print(f"Branch '{branch}' not found.")
+            return False
+
+
     def __init__(self, gitlab: GitLab, config: Configuration, strict: bool):
         super().__init__("files", gitlab)
         self.config = config
         self.strict = strict
         self.branch_protector = BranchProtector(gitlab, strict)
 
-    def _process_configuration(self, project_and_group: str, configuration: dict):
+    def process_configuration(self, project_and_group: str, configuration: dict):
```

Comment:
why has the name of function changed? the initial _ in `_x_y_` denotes a "private" method